src/common.h

```diff
@@ -277,6 +527,40 @@ extern void get_system_HZ(void);
 #endif
 #endif
 
+/**
+ * Read `filename` containing a single number and return it.
+ *
+ * Open `filename`, read the first line, convert it to a number and store it in result.
+ *
+ * @param filename File to read.
+ * @param result Long to store the number in.
+ * @return 0 on success, -1 if could not open file, -2 if could not read from file.
+ */
 extern int read_single_number_file(const char *filename, unsigned long long *result);
 
+/// \todo ktsaou: your help needed
```

Comment:
The general rule is:

1. The `create` call builds a linked list (more like a tree) of substrings to be matched, each matched as PREFIX, SUFFIX, SUBSTRING or EXACT.
2. The `match` call checks a string against a pattern that has been parsed with `create`.
3. The `free` call frees the memory allocated by `create`.

The whole mechanism supports both positive and negative matches.